catalog/app/containers/Bucket/Queries/requests/athena.ts

```diff
@@ -211,19 +211,21 @@ async function waitForQueryStatus(
   athena: Athena,
   QueryExecutionId: string,
 ): Promise<Athena.QueryExecution | null> {
-  const statusData = await athena.getQueryExecution({ QueryExecutionId }).promise()
-  const status = statusData?.QueryExecution?.Status?.State
-  if (status === 'FAILED' || status === 'CANCELLED') {
-    throw new Error(status)
+  let status: string | undefined = 'QUEUED'
+  let queryExecution = null
+  while (status === 'QUEUED' || status === 'RUNNING') {
+    // NOTE: await is used to intetionally pause loop and make requests in series
+    // eslint-disable-next-line no-await-in-loop
+    await wait(1000)
```

Comment:
looks good, but i'd place the timeout after the request, so that the first request is fired right away, and subsequent ones -- following a delay